audio-worklet/design-pattern/lib/em-es6-module.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+// EXPORT_ES6 option does not work as described at
+// https://github.com/kripken/emscripten/issues/6284, so we have to
+// manually add this by '--post-js' setting when the Emscripten compilation.
+export default Module;
```

Comment:
This seems to work well when you don't compile with EMSCRIPTEN with the `-s MODULARIZE=1'

Not so, otherwise

Reply:
That's why the Makefile does not use that command line option:
https://github.com/GoogleChromeLabs/web-audio-samples/blob/master/audio-worklet/design-pattern/wasm/Makefile

Are you suggesting the current code does not compile?

Reply:
Sorry, my comment was unclear. The code compiles well as is. 

If I understood correctly, that set of command line options is the only one that guarantees the correct loading of the wasm module to the AudioWorkletGlobalScope?

I have been trying a compilation with ``-s MODULARIZE_INSTANCE=1`` and ``-s EXPORT_NAME="'<module_name>'"`` but I'm getting a DOMException when loading the worklet.

Reply:
AudioWorkletGlobalScope might be missing some APIs that are available in the main global or WorkerGlobalScope.

`--post-js` above was sort of a workaround because `EXPORT_ES6`option did not work for me. See https://github.com/emscripten-core/emscripten/issues/6284.